modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java

```diff
@@ -1083,7 +1103,8 @@ public CodegenParameter fromRequestBody(RequestBody body, Set<String> imports, S
             // defined.
             if (codegenParameter.vendorExtensions != null && codegenParameter.vendorExtensions.containsKey("x-example")) {
                 codegenParameter.example = Json.pretty(codegenParameter.vendorExtensions.get("x-example"));
-            } else {
+            } else if (!codegenParameter.required) {
+                //mandatory parameter use the example in the yaml. if no example, it is also null.
```

Comment:
Can you explain this logic?

If there's an optional parameter, if it's been provided, we can still provide an example if one has been given?

Reply:
Yes. You can provide example, and all will be ok.
The issue is current example code will choose a value for the parameter. If it is optional, it will give None directly, but if required, it can only use the codegenParameter.example value.
So if we set codegenParameter.example null with required, the example will get problem in generating.

Reply:
Yes. You can provide example, and all will be ok.
The issue is current example code will choose a value for the parameter. If it is optional, it will give None directly, but if required, it can only use the codegenParameter.example value.
So if we set codegenParameter.example null with required, the example will get problem in generating.